### ISSUE TITLE
Using the City Map generates a NullPointerException

### ISSUE THREAD

**Issue by Zsar** [Innoxia/liliths-throne-public#907]
Game version: 0.2.10.2
Java version: 1.8u161
OS: Win10E 10.0.17134

Steps to reproduce:
1. Go to library in Lilaya's house
3. use the Browse the Shelves option
2. use the City Map option

Different tries yielded two different stack traces (diverging at the call to method Game.setContent):

**Comment by Zsar** [comment]
```
Exception in thread "JavaFX Application Thread" java.lang.NullPointerException
	at com.lilithsthrone.game.Game.getCharactersTreatingCellAsHome(Game.java:2568)
	at com.lilithsthrone.rendering.RenderingEngine.appendNPCIcon(RenderingEngine.java:1640)
	at com.lilithsthrone.rendering.RenderingEngine.getFullMap(RenderingEngine.java:1385)
	at com.lilithsthrone.game.dialogue.places.dominion.lilayashome.Library$8.getContent(Library.java:448)
	at com.lilithsthrone.game.Game.setContent(Game.java:1651)
	at com.lilithsthrone.controller.MainController.processResponse(MainController.java:968)
	at com.lilithsthrone.controller.MainController.access$5(MainController.java:966)
	at com.lilithsthrone.controller.MainController$2.handle(MainController.java:908)
	at com.lilithsthrone.controller.MainController$2.handle(MainController.java:1)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventFilterRecord.handleCapturingEvent(CompositeEventHandler.java:282)
	at com.sun.javafx.event.CompositeEventHandler.dispatchCapturingEvent(CompositeEventHandler.java:98)
	at com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(EventHandlerManager.java:223)
	at com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(EventHandlerManager.java:180)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchCapturingEvent(CompositeEventDispatcher.java:43)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:52)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$KeyHandler.process(Scene.java:3964)
	at javafx.scene.Scene$KeyHandler.access$1800(Scene.java:3910)
	at javafx.scene.Scene.impl_processKeyEvent(Scene.java:2040)
	at javafx.scene.Scene$ScenePeerListener.keyEvent(Scene.java:2501)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:217)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(GlassViewEventHandler.java:149)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$352(GlassViewEventHandler.java:248)
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(GlassViewEventHandler.java:247)
	at com.sun.glass.ui.View.handleKeyEvent(View.java:546)
	at com.sun.glass.ui.View.notifyKey(View.java:966)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$147(WinApplication.java:177)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by Zsar** [comment]
```
java.lang.NullPointerException
	at com.lilithsthrone.game.Game.getCharactersTreatingCellAsHome(Game.java:2568)
	at com.lilithsthrone.rendering.RenderingEngine.appendNPCIcon(RenderingEngine.java:1640)
	at com.lilithsthrone.rendering.RenderingEngine.getFullMap(RenderingEngine.java:1385)
	at com.lilithsthrone.game.dialogue.places.dominion.lilayashome.Library$8.getContent(Library.java:448)
	at com.lilithsthrone.game.Game.setContent(Game.java:1651)
	at com.lilithsthrone.controller.eventListeners.SetContentEventListener.handleEvent(SetContentEventListener.java:36)
	at com.sun.webkit.dom.EventListenerImpl.fwkHandleEvent(EventListenerImpl.java:129)
	at com.sun.webkit.WebPage.twkProcessMouseEvent(Native Method)
	at com.sun.webkit.WebPage.dispatchMouseEvent(WebPage.java:807)
	at javafx.scene.web.WebView.processMouseEvent(WebView.java:1045)
	at javafx.scene.web.WebView.lambda$registerEventHandlers$32(WebView.java:1168)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3757)
	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3485)
	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1762)
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2494)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:394)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:295)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$353(GlassViewEventHandler.java:432)
	at com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:389)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:431)
	at com.sun.glass.ui.View.handleMouseEvent(View.java:555)
	at com.sun.glass.ui.View.notifyMouse(View.java:937)
	at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at com.sun.glass.ui.win.WinApplication.lambda$null$147(WinApplication.java:177)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by Zsar** [comment]
Alright, seems that the issue is that this map is not supposed to show NPCs but the game tries to display them anyway. The first try to show the map succeeds, then, not due to "being the first time", but only because a PC has not encountered lingering NPCs in Dominion previously (e.g. because they have not yet left the house).

The method RenderingEngine.getFullMap(WorldType, boolean), called in both stack traces, has to be overloaded with a variant not attempting to display characters...

"// It looks messy, but it works..." - evidently it does not. Never does in the long run.

**Comment by Zsar** [comment]
Turns out that I can reproduce this with a fresh game, skipped prologue, without having interacted with any NPC ever and without having used the city map before. That makes this a [Regression] bug.
- To those who find themselves able to: Please label accordingly!

**Comment by Zsar** [comment]
- no longer reproducible in 0.3.2.2 Alpha
- mentioned as fixed in changelog of 0.2.11.9

Closing issue.
